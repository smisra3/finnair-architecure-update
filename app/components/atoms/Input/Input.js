// @flow
import type { Node } from 'react';
import camelCase from 'lodash/camelCase';
import styled from 'styled-components';
import styles from './Input.style';
import FieldError from '../FieldError/FieldError';

type Props = {
  className: string,
  id: string | number,
  labelContent?: string,
  inheritedStyles?: string,
  errorMessage?: string,
  maxLength?: number,
  onKeyUp?: Function,
  iconClickHandler?: Function,
  inputRef?: Function,
  type?: string,
  placeholder?: string,
  value?: string,
  required?: boolean,
  dataSlnmId?: string,
  onChange?: Function
};

const keyUpHandler = (event, maxLength) => {
  if (event.target.value.length >= parseInt(maxLength, 10)) {
    // eslint-disable-next-line
    event.target.value = event.target.value.substring(0, parseInt(maxLength, 10));
  }
};

const Input = ({
  className,
  id,
  labelContent,
  errorMessage,
  inheritedStyles,
  maxLength,
  onKeyUp,
  iconClickHandler,
  inputRef,
  type,
  placeholder,
  value,
  onChange,
  required,
  dataSlnmId,
  labelAddOnComponent,
  ...others
}: Props): Node => {
  return (
    <div className={className}>
      <div className="label-wrapper">
        <label htmlFor={id}>{labelContent}</label>
        {labelAddOnComponent ? (<div className="addOnComponentWrapper">{labelAddOnComponent}</div>) : null}
      </div>
      <div className="input-wrapper">
        <input
          id={id}
          maxLength={maxLength}
          type={type}
          onKeyUp={(event) => {
            keyUpHandler(event, maxLength);
            if (onKeyUp) onKeyUp(event);
          }}
          onChange={(event) => {
            if (onChange) onChange(event);
          }}
          onBlur={(event) => {
            if (onBlur) onBlur(event);
          }}
          ref={inputRef}
          placeholder={placeholder}
          value={value}
          aria-invalid={errorMessage && errorMessage.length ? true : null}
          aria-required={required || null}
          data-slnm-id={camelCase(placeholder)}
          {...others}
        />
      </div>
      <FieldError errorMessage={errorMessage} dataSlnmId={dataSlnmId} />
    </div>
  )
}

Input.defaultProps = {
  type: 'text',
  labelContent: '',
  inheritedStyles: '',
  errorMessage: '',
  maxLength: 100,
  onKeyUp: () => { },
  iconClickHandler: () => { },
  inputRef: () => { },
  placeholder: '',
  value: '',
  required: false,
  dataSlnmId: 'InputField',
  onChange: () => { },
  onBlur: () => { },
  isParentSubmitted: false
};

export default styled(Input)`
  ${styles};
`;
export { Input as InputVanilla };
