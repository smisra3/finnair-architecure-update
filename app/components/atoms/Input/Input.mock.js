import { action } from '@storybook/addon-actions';
import defaultAnchor from '../Anchor/Anchor.mock';
import Anchor from '../Anchor/index';

const textInput = {
  className: 'example',
  labelContent: 'User Name',
  type: 'text',
  required: true,
  isError: false,
  onChange: action('changed'),
};

const checkboxInput = {
  className: 'example',
  labelContent: 'User Name',
  type: 'checkbox',
  required: true,
  isError: false,
  onChange: action('changed'),
  value: '',
};

const textInputWithCTA = {
  className: 'example',
  labelContent: 'Username',
  type: 'text',
  required: true,
  isError: false,
  onChange: action('changed'),
  labelAddOnComponent: (<Anchor {...defaultAnchor} >Forgot your username</Anchor>),
};


const textInputWithError = {
  className: 'example',
  labelContent: 'User Name',
  type: 'text',
  required: true,
  isError: true,
  errorMessage: 'Please enter your username or CRN',
  onChange: action('changed'),
};

export {
  textInput,
  textInputWithError,
  textInputWithCTA,
  checkboxInput,
};
