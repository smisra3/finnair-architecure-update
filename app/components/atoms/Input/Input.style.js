import { css } from 'styled-components';

const commonStyles = css`
  .label-wrapper {
    padding-bottom: 5px;
    width: 100%;
    position: relative;
    ${props => (props.labelAddOnComponent ? 'display:flex;' : '')};

    & label {
      height: 100%;
      font-weight: ${props => props.theme.fontBold};
      font-size: ${props => props.theme.fontSizeDefault};
      color: ${props => props.theme.textColorSecondary};
      margin: 0;
      ${props => (props.labelAddOnComponent ? 'flex: 1 1 auto;' : '')};
    }

    & .addOnComponentWrapper {
      flex: 0 0 auto;
      height: 100%;
      font-size: ${props => props.theme.fontSizeDefault};
    }
  }

  input {
    width: 100%;
    height: 18px;
    -webkit-tap-highlight-color: transparent;
    -webkit-appearance: none;
    -moz-appearance: none;
    border-radius: 0;
    font-size: ${props => props.theme.fontSizeBase};
    background-color: ${props => props.theme.backgroundColorPrimary};
    height: 100%;
    color: ${props => props.theme.textColorSecondary};
    padding: 7px 8px;
    border: 1px solid;
    box-shadow: none;
    border-color: ${props => (props.errorMessage ? props.theme.fieldBorderErrorColor : props.theme.fieldBorderColor)};
    &::placeholder {
      color: ${props => props.theme.textColorPlaceholder};
    }

    &::-ms-clear {
      display: none;
    }

    &:disabled {
      background: ${props => props.theme.backgroundColorDisabled};
      color: ${props => props.theme.textColorPlaceholder};
      border: 1px solid ${props => props.theme.fieldBorderDisabledColor};
    }

    &:focus {
      outline: none;
      border-color: ${props => (props.errorMessage ? props.theme.fieldBorderErrorColor : props.theme.fieldBorderFocusColor)};
    }
  }

  & .input-wrapper{
    margin-bottom: 10px;
  }

  input:-webkit-autofill {
    background-color: ${props => props.theme.fieldBorderColor};
  }
`;

export default commonStyles;
