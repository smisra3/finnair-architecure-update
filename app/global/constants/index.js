/* TODO: Remove lint exception */
/* eslint-disable import/prefer-default-export */
export const DEVICE_TYPE = 'DEVICE_TYPE';
export const IS_TABLET = 'IS_TABLET';
export const REGION_TYPE = 'REGION_TYPE';
export const CONFIG_KEYS = 'CONFIG_KEYS';
export const CURRENT_ROUTE = 'CURRENT_ROUTE';
export const PAGE_URL = 'PAGE_URL';
export const PAGE_QUERY = 'PAGE_QUERY';
export const GLOBAL_DATA_FAILURE = 'FIL/GLOBAL/GLOBAL_DATA_FAILURE';
export const CHECK_AUTH = 'FIL/GLOBAL/CHECK_AUTH';
export const CHECK_AUTH_SUCCESS = 'FIL/GLOBAL/CHECK_AUTH_SUCCESS';
export const SET_USER_STATE = 'FIL/GLOBAL/SET_USER_STATE';
export const PAGE_ORIGIN = 'PAGE_ORIGIN';
export const LABELS_STATE_KEY = 'labels';

export const USER_STATE_LOGGED_IN = 'FIL/GLOBAL/SUCCESS';
export const USER_STATE_RECOGNIZED = 'FIL/GLOBAL/RECOGNIZED';
export const USER_STATE_ANONYMOUS = 'FIL/GLOBAL/ANONYMOUS';

export const GET_LABELS = 'FIL/GLOBAL/GET_LABELS';
export const GET_LABELS_SUCCESS = 'FIL/GLOBAL/GET_LABELS_SUCCESS';
export const GET_LABELS_FAILURE = 'FIL/GLOBAL/GET_LABELS_FAILURE';
