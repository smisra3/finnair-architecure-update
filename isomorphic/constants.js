const HEARTBEAT_SESSION_NAME = 'JSESSIONID';
const ENV_DEVELOPMENT = 'development';
const ENV_PRODUCTION = 'production';
const SERVICE_WORKER_FILE = '/service-worker.js';
const API_PROXY_PATH = '/api';
const CLEAR_SERVICES_CACHE = '/clear-services-cache';
const HEALTH_CHECK = '/healthcheck';
const PROTOCOL_HTTPS = 'https';
const RESPONSE_OK = 200;
const RESPONSE_INTERNAL_SERVER_ERROR = 500;
const STATIC_FOLDER_PREFIX = '/static/';

module.exports = {
  HEARTBEAT_SESSION_NAME,
  ENV_DEVELOPMENT,
  ENV_PRODUCTION,
  SERVICE_WORKER_FILE,
  API_PROXY_PATH,
  CLEAR_SERVICES_CACHE,
  PROTOCOL_HTTPS,
  RESPONSE_OK,
  RESPONSE_INTERNAL_SERVER_ERROR,
  STATIC_FOLDER_PREFIX,
  HEALTH_CHECK,
};
