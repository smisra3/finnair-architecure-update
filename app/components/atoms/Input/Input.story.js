// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  textInput,
  textInputWithCTA,
  textInputWithError,
} from './Input.mock';

// Import Vanilla Component to display markup and props
import { InputVanilla } from './Input';
// Import Styled Component to showcase variations
import Input from './index';

const wrapperStyle = {
  width: '400px'
};

storiesOf('Atoms', module).addWithChapters('Inputs', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <InputVanilla {...textInput} className={`hide-default-sample ${textInput.className}`} />,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Input Variations',
      sections: [
        {
          title: 'Default Text Box',
          sectionFn: () => <Input {...textInput} />,
        },
        {
          title: 'Text Box with CTA',
          sectionFn: () => <Input {...textInputWithCTA} />,
        },
        {
          title: 'Text Box With Inline Error',
          sectionFn: () => <Input {...textInputWithError} />,
        },
        {
          title: 'Text Box inside a wrapper of fixed width',
          sectionFn: () => (<div style={wrapperStyle}><Input {...textInputWithError} /></div>),
        },
      ],
    },
  ],
});
