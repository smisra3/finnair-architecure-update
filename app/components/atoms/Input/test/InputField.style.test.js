import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Input from '../';
import Theme from '../../../../styles/theme';
import { DESKTOP } from '../../../../constants';

const theme = {
  ...Theme,
  deviceType: DESKTOP,
};

describe('Input', () => {
  let input = '';
  const props = {
    deviceType: DESKTOP,
  };

  beforeEach(() => {
    input = renderer.create(<Input id="test" {...props} theme={theme} />).toJSON();
  });

  test('should render correctly', () => {
    expect(input).toMatchSnapshot();
  });
});
