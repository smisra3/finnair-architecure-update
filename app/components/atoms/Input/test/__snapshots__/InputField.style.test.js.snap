// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
.c0 .label-wrapper {
  padding-bottom: 8px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c0 .label-wrapper label {
  height: 100%;
  font-weight: 700;
  font-size: 12px;
  color: #262626;
}

.c0 .label-wrapper .addOnComponentWrapper {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  height: 100%;
  font-size: 12px;
}

.c0 input {
  width: 100%;
  -webkit-tap-highlight-color: transparent;
  background-color: #fff;
  height: 100%;
  color: #262626;
  padding: 7px 8px;
  border: 1px solid;
  box-shadow: none;
  border-color: #bfbfbf;
}

.c0 input::-webkit-input-placeholder {
  color: #dfdfdf;
}

.c0 input::-moz-placeholder {
  color: #dfdfdf;
}

.c0 input:-ms-input-placeholder {
  color: #dfdfdf;
}

.c0 input::placeholder {
  color: #dfdfdf;
}

.c0 input::-ms-clear {
  display: none;
}

.c0 input:disabled {
  background: grey;
  color: #dfdfdf;
  border: 1px solid #bfdfdf;
}

.c0 input:focus {
  outline: none;
  border-color: #006193;
}

.c0 input:-webkit-autofill {
  background-color: #bfbfbf;
}

<div
  className="c0"
>
  <div
    className="label-wrapper"
  >
    <label
      htmlFor="test"
    >
      
    </label>
  </div>
  <div
    className="input-wrapper"
  >
    <input
      aria-invalid={null}
      aria-required={null}
      data-slnm-id=""
      deviceType="desktop"
      id="test"
      maxLength={100}
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder=""
      theme={
        Object {
          "backgroundColorDisabled": "grey",
          "backgroundColorPrimary": "#fff",
          "backgroundColorSecondary": "#eaf2f6",
          "backgroundColorSuccess": "#2e993a",
          "brandPrimaryColor": "#006193",
          "buttonBackgroundColorPrimary": "#e57725",
          "buttonBackgroundColorSecondary": "#016fad",
          "buttonColorPrimary": "#ffffff",
          "buttonColorSecondary": "#ffffff",
          "cautionBackgroundColor": "#fffdf3",
          "colorOutline": "#999",
          "containerShadowPrimary": "rgba(0,0,0,.2)",
          "deviceType": "desktop",
          "fieldBorderColor": "#bfbfbf",
          "fieldBorderDisabledColor": "#bfdfdf",
          "fieldBorderErrorColor": "#c00",
          "fieldBorderFocusColor": "#006193",
          "fontBold": "700",
          "fontBook": "400",
          "fontFamilyDefault": "Arial, Helvetica, sans-serif",
          "fontFamilyIcon": "",
          "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
          "fontFamilySecondary": "Arial, Helvetica, sans-serif",
          "fontMedium": "500",
          "fontNormal": "normal",
          "fontSemiBold": "600",
          "fontSizeBase": "16px",
          "fontSizeButtonPrimary": "24px",
          "fontSizeButtonSecondary": "16px",
          "fontSizeDefault": "12px",
          "letterSpacing": "1px",
          "lineHeightBase": "1.5",
          "linkColorPrimary": "#008CBA",
          "textColorError": "#c00",
          "textColorPlaceholder": "#dfdfdf",
          "textColorPrimary": "#222",
          "textColorSecondary": "#262626",
          "textColorSuccess": "#2e993a",
          "zIndex": Object {
            "zFilter": 100,
            "zFilterActions": 400,
            "zFilterList": 150,
            "zFlyout": 200,
            "zLoader": 1000,
            "zMenu": 800,
            "zNativeTickSelectIcon": 50,
            "zOverlay": 900,
            "zPlacesAutocomplete": 1100,
            "zSticky": 700,
          },
        }
      }
      type="text"
      value=""
    />
  </div>
  
</div>
`;
